### PR TITLE
ISPN-15225 Server Footprint

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-15225

### CODE REVIEW

**Comment by alanfx** [comment]
Hey @tristantarrant should these values be tracked over time in Horreum, ReportPortal, or something else to see when the values change? Just curious how this test result will be used.

**Comment by tristantarrant** [comment]
I wasn't thinking of keeping track of this in other systems. The purpose of this test is to ensure that we stay within reasonable upper/lower bounds for both metrics. If the test fails during a PR build, then the author will have to analyse the reason and, if necessary, readjust the baseline values. Having both lower and upper bounds allows us to reduce the baseline.

**Comment by alanfx** [comment]
Ok, makes sense. Thanks for the explanation!

**Comment by pruivo** [comment]
Ironically, the test is already failing :rofl: 
https://ci.infinispan.org/job/Infinispan/job/PR-11374/1/testReport/junit/org.infinispan.server.footprint/FootprintIT/testFootprint/

**Comment by wburns** [comment]
Now that RESP segment consolidation is complete this needs to be rebased to see the new values.

**Comment by tristantarrant** [comment]
I have updated the values based on the following numbers:
Loaded classes: 10990
Heap: 25330368

**Comment by tristantarrant** [comment]
I have also added a disk footprint test using the same approach as the memory one (with lower/upper bounds). 
I've relaxed the memory upper bound until Log4j 2.22.0 is out.

File: server/runtime/pom.xml [hunk 1]
```diff
@@ -283,6 +283,10 @@
          <artifactId>junit</artifactId>
          <scope>test</scope>
       </dependency>
+      <dependency>
+         <groupId>org.assertj</groupId>
+         <artifactId>assertj-core</artifactId>
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply server/runtime/pom.xml]: I've moved the disk footprint test to `FootprintIT` so this is no longer needed.

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Pull request merged**